src/glyph_item_iter.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright 2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib::translate::*;
+use pango_sys;
+use GlyphItem;
+
+//Note: This type not exported
+glib_wrapper! {
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct GlyphItemIter(Boxed<pango_sys::PangoGlyphItemIter>);
+
+    match fn {
+        copy => |ptr| pango_sys::pango_glyph_item_iter_copy(mut_override(ptr)),
+        free => |ptr| pango_sys::pango_glyph_item_iter_free(ptr),
+        init => |_ptr| (),
+        clear => |_ptr| (),
+        get_type => || pango_sys::pango_glyph_item_iter_get_type(),
+    }
+}
+
+impl GlyphItemIter {
+    pub fn init_end(glyph_item: &GlyphItem, text: &str) -> Option<GlyphItemIter> {

```

Comment:
This is still unsafe: you need to ensure that the parameters stay alive as long as the return iter. And you need to ensure that the text matches the glyph item, otherwise invalid memory accesses will happen.

Reply:
This type not exported, I can change to `pub(crate)` to be more explicit,
and only used within lifetimed GlyphItemIterator

Reply:
Generally, if soundness of a function depends on its usage (i.e. it doesn't guarantee), it should be marked `unsafe` regardless of whether it's internal or not.

Reply:
Oh I missed that. Better make it more explicit then, yes :)

Reply:
> Generally, if soundness of a function depends on its usage (i.e. it doesn't guarantee), it should be marked unsafe regardless of whether it's internal or not.

Yes, definitely. but the function is also gone now, so... :)